zerver/lib/actions.py

```diff
@@ -843,14 +843,18 @@ def do_set_realm_property(
             return
 
         user_profiles = UserProfile.objects.filter(realm=realm, is_bot=False)
+
+        # TODO: Design a bulk event for this or force-reload all clients
         for user_profile in user_profiles:
             user_profile.email = get_display_email_address(user_profile)
-            # TODO: Design a bulk event for this or force-reload all clients
-            send_user_email_update_event(user_profile)
-        UserProfile.objects.bulk_update(user_profiles, ["email"])
+            # We need to make sure we bind the event to the
+            # current value of user_profile, no the loop variable.
+            transaction.on_commit(send_user_email_update_event_lambda(user_profile))
```

Comment:
Missing `lambda:`.

Reply:
Oh I see, the lambda is buried inside the function. (I don’t think there’s a reason to wrap it like that, though.)